### ISSUE TITLE
How to use with rooms?

### ISSUE THREAD

**Issue by javajohnHub** [share/sharedb#150]
I have a socket.io chat application with rooms. 
When a user joins a room I want to have a collaborative text-editor using sharedb.
How can I pass the room to sharedb to listen to?

**Comment by pedrosanta** [comment]
I think ShareDB requires a WebSocket API-compatible transport layer, and I think Socket.io doesn't provide that. You would have to code a wrapper - I tried to use Socket.io a while ago, but ended up using sockets with reconnection (check #121 to know more). Also, I think there might be issues with 'long-polling' transport of Socket.io/Engine.io that can produce scenarios where out-of-order message delivery can occur (this would still have to be confirmed), so even when trying to use Socket.io the connection upgrade should be disabled and only use the 'socket' transport.

**TL;DR;:** Socket.io connection/transport layer isn't supported and remains to be thoroughly tested. Check #121 for alternatives.

**Comment by curran** [comment]
It seems this issue can be closed.

**Comment by alecgibson** [comment]
Closing due to inactivity.

**Comment by jamesopti** [comment]
For the posterity of others who arrive here trying to use SocketIO on the same server as sharedb, I came across this issue and solved it using 2 separate socket instances, one SocketIO and the other a native WS one for sharedb using the `noServer` option ([example from the docs](https://github.com/websockets/ws#multiple-servers-sharing-a-single-https-server)):


```ts
...
const server = new Server(expressApp)
const shareDBSocketServer = new WebSocket.Server({ noServer: true })
const socketIOSocketServer = new SocketIOServer(server, {
  path: '/otherservice',
  cors: { /* ... */  },
})
server.on('upgrade', function upgrade(request, socket, head) {
  // socketIOSocketServer upgrades are handled by express
  if (!request.url.startsWith('/otherservice')) {
    shareDBSocketServer.handleUpgrade(request, socket, head, function done(ws) {
      shareDBSocketServer.emit('connection', ws, request)
    })
  }
})
shareDBSocketServer.on('connection', function connection(ws: WS.Socket) {
  // ... init shareDB
});

socketIOSocketServer.on('connection', function connection(io: SocketIO.Socket) {
  // ... do other service stuff with socket.io
});
```
